### PR TITLE
Monerodart -> testing integration

### PR DESCRIPTION
**Pull request by sneurlax**
Reference #838; basically #838 + `testing` branch (which includes Bitcoin Frost and other changes)

### CODE REVIEW

File: android/app/src/main/jniLibs/arm64-v8a/libmonero_libwallet2_api_c.so
Comment (julian-CStack) [inline android/app/src/main/jniLibs/arm64-v8a/libmonero_libwallet2_api_c.so:1]: delete

File: android/app/src/main/jniLibs/armeabi-v7a/libmonero_libwallet2_api_c.so
Comment (julian-CStack) [inline android/app/src/main/jniLibs/armeabi-v7a/libmonero_libwallet2_api_c.so:1]: delete

File: android/app/src/main/jniLibs/x86_64/libmonero_libwallet2_api_c.so
Comment (julian-CStack) [inline android/app/src/main/jniLibs/x86_64/libmonero_libwallet2_api_c.so:1]: delete

File: scripts/android/build_all.sh [hunk 1]
```diff
@@ -12,11 +12,11 @@ mkdir -p build
 
 PLUGINS_DIR=../../crypto_plugins
 
-(cd "${PLUGINS_DIR}"/flutter_liblelantus/scripts/android && ./build_all.sh ) &
-(cd "${PLUGINS_DIR}"/flutter_libepiccash/scripts/android && ./build_all.sh )  &
-(cd "${PLUGINS_DIR}"/flutter_libmonero/scripts/android/ && ./build_all.sh  ) &&
-set_rust_to_1720 &&
-(cd "${PLUGINS_DIR}"/frostdart/scripts/android && ./build_all.sh ) &
+(cd "${PLUGINS_DIR}"/flutter_liblelantus/scripts/android && ./build_all.sh )
+(cd "${PLUGINS_DIR}"/flutter_libepiccash/scripts/android && ./install_ndk.sh && ./build_openssl.sh && ./build_all.sh )
```
Comment (julian-CStack) [inline scripts/android/build_all.sh:16]: libepiccash's build all script runs install ndk and openssl (or at least it did last I checked)

File: macos/wownero_libwallet2_api_c.dylib
Comment (julian-CStack) [inline macos/wownero_libwallet2_api_c.dylib:1]: delete

File: macos/monero_libwallet2_api_c.dylib
Comment (julian-CStack) [inline macos/monero_libwallet2_api_c.dylib:1]: delete

File: lib/wallets/wallet/impl/wownero_wallet.dart [hunk 2]
```diff
@@ -330,7 +337,9 @@ class WowneroWallet extends CryptonoteWallet with CwBasedInterface {
         //     date: DateTime.now().subtract(const Duration(
         //         days:
         //         2))); // subtract a couple days to ensure we have a buffer for SWB
-        final bufferedCreateHeight = getSeedHeightSync(wallet!.seed.trim());
+        // TODO(mrcyjanek): implement
```
Comment (julian-CStack) [inline lib/wallets/wallet/impl/wownero_wallet.dart:340]: needs to be done asap

File: lib/wallets/wallet/impl/wownero_wallet.dart [hunk 3]
```diff
@@ -441,7 +450,8 @@ class WowneroWallet extends CryptonoteWallet with CwBasedInterface {
 
         // extract seed height from 14 word seed
         if (seedLength == 14) {
-          height = getSeedHeightSync(mnemonic.trim());
+          // TODO(mrcyjanek): implement
+          height = 1; // getSeedHeightSync(mnemonic.trim());
```
Comment (julian-CStack) [inline lib/wallets/wallet/impl/wownero_wallet.dart:454]: needs to be done asap

File: .fvmrc
Comment (julian-CStack) [inline .fvmrc:1]: delete (otherwise we need to update this every time we update the flutter version)

File: .vscode/settings.json
Comment (julian-CStack) [inline .vscode/settings.json:1]: revert this change

File: .gitignore
Comment (julian-CStack) [inline .gitignore:1]: keeping this won't hurt since we don't need to update this

File: android/app/src/main/AndroidManifest.xml
Comment (julian-CStack) [inline android/app/src/main/AndroidManifest.xml:1]: Any reason extractNativeLibs is specifically set here?
If there is no specific reason then I suggest reverting and letting the default be chosen by the system as described in: https://developer.android.com/guide/topics/manifest/application-element

File: android/app/build.gradle
Comment (julian-CStack) [inline android/app/build.gradle:1]: Reasons for this?

File: crypto_plugins/flutter_libmonero
Comment (julian-CStack) [inline crypto_plugins/flutter_libmonero:1]: This requires a separate review of the changes to flutter_libmonero

**Review by julian-CStack — CHANGES_REQUESTED** [review]
see comments

File: android/app/src/main/AndroidManifest.xml
Comment (MrCyjaneK) [reply android/app/src/main/AndroidManifest.xml:1]: for the .so to be extracted, the ones that are marked for deletion. I'll make it a cmake install step instead if symlinking is not desired

File: android/app/build.gradle
Comment (MrCyjaneK) [reply android/app/build.gradle:1]: monero_c also depends on libc++_shared.so, and in early versions of this PR i was getting build errors because multiple packages were providing the dynamic library

File: scripts/android/build_all.sh (hunk 1 above)
Comment (sneurlax) [reply scripts/android/build_all.sh]: Yes it does.  Resolved by ce17fed2bdc21147ed21164d505dd01d1feac2fbe4420f8c561ba6cd0da91607

File: .gitignore
Comment (sneurlax) [reply .gitignore:1]: but I use fvm on macos :P

File: .vscode/settings.json
Comment (sneurlax) [reply .vscode/settings.json:1]: Resolved by 89a687b9715a8a69590f01da76b83c2a8424c4ee

**Comment by MrCyjaneK** [comment]
@julian-CStack regarding .so/.dylib symlinks deletion, what is the preferred way to include them in the apk/.app on android/macos?

File: lib/wallets/wallet/impl/monero_wallet.dart [hunk 4]
```diff
@@ -218,8 +277,8 @@ class MoneroWallet extends CryptonoteWallet with CwBasedInterface {
           final addressString =
               (CwBasedInterface.cwWalletBase as MoneroWalletBase?)
                   ?.getTransactionAddress(
-            addressInfo!['accountIndex'] as int,
-            addressInfo['addressIndex'] as int,
+            addressInfo?['accountIndex'] as int? ?? 0,
+            addressInfo?['addressIndex'] as int? ?? 0,
```
Comment (julian-CStack) [inline lib/wallets/wallet/impl/monero_wallet.dart:281]: Is it ok to default to 0 here? Or should there be some error?

Comment (MrCyjaneK) [reply lib/wallets/wallet/impl/monero_wallet.dart:281]: should be fine, there is no account switching feature in stack yet.

**Comment by julian-CStack** [comment]
merged https://github.com/cypherstack/stack_wallet/tree/monerodart_testing into https://github.com/cypherstack/stack_wallet/tree/monero_merge which is based on current staging that includes a pile of generalization changes
